### DISCUSSION TITLE
Customizing Beamer's theme

### DISCUSSION THREAD

**Discussion by TUD-ADS** [tud-cd/tud-cd#35] [General]
Hello,

I would like to make some modifications to the beamer's theme to adapt it to our chair and as the theme is so complex, I would like some help if possible to point me where I should make the changes.

* Title page:
	* Move the title to the first position, keeping the font and size it has now.
	* Include under the Author's name: Institute of X / Chair of Y.
	* Option to add chair's logo next to the DDC one (maybe with option to the left or right to be a parameter) or option to include own file for zweitlogo rather than hardcoding it
* Option to have the footer as header. If header:
	* frametitle in between the logos
	* Include a second gray line to have in between both lines the sections
	* Footer:
		* Left: Author's name
		* Center: Title of presentation or short title
		* Right: Slide X of Y

Any kind of help will be much appreciated.
Best regards,
Ariel.

**Comment by keinstein** [comment]
According to your explanations, I think, your chair is still using the outdated Corporate Design that has been replaced in the beginning of 2018 by the new design. Your chair should update its templates to the ones from  [the official CD site](https://tu-dresden.de/intern/services-und-hilfe/kommunizieren-und-publizieren/cd). They addressed many of the drawbacks of the old CD.

You might take a look at the deprecated compatibility option `cd2012ts` to the beamer theme.

Institutional information is set with
```
\einrichtung{School}
\fachrichtung{Faculty}
\institut{Institute of XXX}
\professur{Chair of YYY}
```
in that order. 

Additional logos had been forbidden by the corporate design short after its introduction. So you should negotiate with the Corporate Design Office about the current rules. Nevertheless you can configure them with `\setbeamertemplate{zweitlogo/headline}[logofile]{fremdfirma.png}` or `\setbeamertemplate{zweitlogo/headline}[freeform]{\LaTeX}`.

Regarding your other suggestions: You can probably implement them based on `cd2012ts`, which is the old CD with my modifications. If you do modifications, please make them optional, so that the default still follows the official Corporate Design either in form of the Corporate Design Handbook or via negotiations with the Corporate Design office of TU Dresden. Pull requests are always welcome, even if they are strictly speaking not CD conform. LaTeX provides some solutions that are not possible with PowerPoint in that way. And it would help all TU employees, if all CDs are fed from a single source that works together with the CD office.

**Comment by TUD-ADS** [comment]
We will update to the new template rather than doing the modifications then. However, I am not able to compile when I add the institutional information and I get errors `"Undefined control sequence"` for `fachrichtung`, `institut`, and `professur`. Do I also need to add `einrichtung` or can I omit it? I feel I need to add something else in order to add them, right?

```
\documentclass{beamer}

\title{Title}
\author{Author}
\fachrichtung{Faculty}
\institut{Institute of XXX}
\professur{Chair of YYY}
\usetheme[section,pagenum]{tud}

\begin{document}

\maketitle

\section{First}
\begin{frame}
\frametitle{Introduction}
\begin{itemize}
    \item A\pause
    \item B
\end{itemize}
\end{frame}

\begin{frame}
\frametitle{Motivation}
\begin{itemize}
    \item A
    \item B
\end{itemize}
\end{frame}

\section{Second}
\begin{frame}
\frametitle{Contribution 1}
\begin{itemize}
    \item A
    \item B
\end{itemize}
\end{frame}

\begin{frame}
\frametitle{Contribution 2}
\begin{itemize}
    \item A
    \item B
\end{itemize}
\end{frame}

\end{document}
```
Also the sections are not shown on the slides. And lastly, is there any option for short title? Because a long one, which would only be displayed in the title page, would not fit in the footer.
Thanks for the help.
